theia-tree-editor/src/browser/tree-editor-widget.tsx

```diff
@@ -160,7 +161,7 @@ export abstract class BaseTreeEditorWidget extends BaseWidget implements Saveabl
     protected abstract handleFormUpdate(
         data: any,
         node: TreeEditor.Node
-    ): void;
+    ): MaybePromise<void>;
```

Comment:
I'd just make this a normal `Promise`. It makes sense that this is handled asynchronously anyway.